data_import/views.py

```diff
@@ -34,7 +34,17 @@ def get(self, request, *args, **kwargs):
         if unavailable:
             return HttpResponseForbidden('<h1>This file is unavailable.</h1>')
 
-        return super(DataFileDownloadView, self).get(request, *args, **kwargs)
+        query_key = request.GET.get('key', None)
+        if not self.data_file.is_public:
+            if not query_key:
+                return HttpResponseForbidden('<h1>No key provided.</h1>')
```

Comment:
Do we expect people to see these? Wondering whether it might be worthwhile to throw a proper error with the response instead of giving it a human-readable message? Or do I miss how HttpResponseForbidden works? :)

Reply:
Well, we could go with this:
https://docs.djangoproject.com/en/2.1/ref/views/#the-403-http-forbidden-view

I was adding to the already existing behaviour.  I feel it's nice to say why things are failing, which is the nice thing about returning something with HttpResponseForbidden.

Honestly, I could go either way.

Reply:
Related docs on HttpResponseForbiden:
https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpResponseForbidden

If you think we'd be better off throwing a generic 403, then probably rewriting the entire method to be consistent would be a good idea.